Mathlib/Computability/PrimrecFilter.lean

```diff
@@ -0,0 +1,209 @@
+/-
+Copyright (c) 2025 David J. Webb. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David J. Webb
+-/
+import Mathlib.Computability.Primrec
+/-!
+
+# Bounded quantifiers are primitive recursive
+This file contains lemmata for showing bounded existentially and universally quantified
+statements are primitive recursive, as well as more general filtering for arbitrary
+primcodable types.
+
+## Main results
+- Filtering for elements from a list that meet a primrec criterion is primrec
+- Checking whether a list has some element meeting a primrec criterion is primrec
+- Checking whether every element of a list meets a primrec criterion is primrec
+- Primitive recursive functions are closed under bounded existential and universal quantifiers
+
+## References
+- [R. I. Soare *Turing Computability - Theory and Applications*] [Soare2016]
+-/
+
+open List
+open Primrec
+
+namespace Primrec
+
+/-- Filtering a list for elements that satisfy a decidable predicate is primitive recursive. -/
+lemma list_filter {α} [Primcodable α] (f : α → Prop) [DecidablePred f]
+(hf : PrimrecPred f) : Primrec fun L => (filter (fun a => f a) L) := by
+  rw [← List.filterMap_eq_filter]
+  apply listFilterMap Primrec.id
+  simp [Primrec₂, Option.guard]
```

Comment:
This is a non-terminal simp: this simp does not close the goal. These make maintaining the library harder in the long run, as this could break if new simp lemmas are added. Can you replace it by a suitable `simp only` call, please?
If you write `simp?`, you get a suggestion what you can replace it with.

Another option is to write `simpa [Primrec₂, Option.guard] using ite (PrimrecPred.comp hf snd) (option_some_iff.mpr snd) (const none)`, that also works.